bundles/config/org.eclipse.smarthome.config.dispatch/src/main/java/org/eclipse/smarthome/config/dispatch/internal/ConfigDispatcher.java

```diff
@@ -151,6 +163,62 @@ protected void processWatchEvent(WatchEvent<?> event, Kind<?> kind, Path path) {
             } catch (IOException e) {
                 logger.warn("Could not process config file '{}': {}", path, e);
             }
+        } else if (kind == ENTRY_DELETE) {
+            // Detect if a service specific configuration file was removed. We want to
+            // notify the service in this case with an updated empty configuration.
+            File configFile = path.toFile();
+            if (configFile.isHidden() || configFile.isDirectory() || !configFile.getName().endsWith(".cfg")) {
+                return;
+            }
+
+            exclusivcePIDMap.setFileRemoved(configFile.getAbsolutePath());
+            processOrphanExclusivePIDs();
+            storeCurrentExclusivePIDList();
+        }
+    }
+
+    private void loadExclusivePIDList() {
+        try (FileReader reader = new FileReader(exclusivePIDStore)) {
+            exclusivcePIDMap = GSON.fromJson(reader, ExclusivePIDMap.class);
+            if (exclusivcePIDMap != null) {
+                exclusivcePIDMap.initializeProcessPIDMapping();
+            }
+        } catch (JsonSyntaxException | JsonIOException | IOException e) {
+            logger.error("Error loading exclusive pids from {}.", exclusivePIDStore.getAbsolutePath(), e);
+        } finally {
+            if (exclusivcePIDMap == null) {
+                exclusivcePIDMap = new ExclusivePIDMap();
+            }
+        }
+    }
+
+    private void storeCurrentExclusivePIDList() {
+        try (FileWriter writer = new FileWriter(exclusivePIDStore)) {
+            exclusivcePIDMap.setCurrentExclusivePIDList();
+            GSON.toJson(exclusivcePIDMap, writer);
+        } catch (JsonIOException | IOException e) {
+            logger.error("Error storing exclusive PID list in bundle data file.", e);
+        }
+    }
+
+    private void processOrphanExclusivePIDs() {
+        for (String orpahnPID : exclusivcePIDMap.getOrpahnPIDs()) {
+            try {
+                Configuration configuration = configAdmin.getConfiguration(orpahnPID, null);
+                configuration.delete();

```

Comment:
Is this a "silent" operation, if the configuration didn't exist at all?

Reply:
In line 207 the configuration object is created if none existed before. So with regards to object references this will always proceed. I´m not sure what "didn't exist at all" could otherwise mean in OSGi context.

Reply:
I meant if we are sure that we are not causing an "config CREATED" and "config DELETED" event on ConfigAdmin for no reason.